File: src/main/resources/static/app/controller/design.controller.js
/**
 * Created by Sakshi Kukreti on 27-04-2017.
 */

angular.module('app')
    .controller('designController', function ($scope, designService) {
        $scope.saveChanges = function () {
            $scope.showLoader = true;
            designService
                .saveChanges($scope.designs)
                .then(function (response) {
                    $scope.showLoader = false;
                    console.log('updating design', response);
                    initController();
                });
        };

        function initController() {
            $scope.showLoader = true;
            designService.fetchAllDesigns()
                .then(function (response) {
                    $scope.showLoader = false;
                    console.log(response);
                    $scope.designs = response.data;
                });

        }

        initController();
    });

File: src/main/java/com/example/webservice/controller/DesignController.java
package com.example.webservice.controller;

import com.example.webservice.domain.Design;
import com.example.webservice.repository.DesignRepository;
import com.example.webservice.service.DesignService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * Created by Sakshi Kukreti on 26-04-2017.
 */
@RestController
@RequestMapping(value = "service")
public class DesignController {

    @Autowired
    private DesignRepository designRepository;

    @Autowired
    private DesignService  designService;

    @RequestMapping(value = "/{designNum}/{author}", method = RequestMethod.GET, produces = "application/json")
    public List<Design> getDesignByDesignNumAndAuthor(@PathVariable("designNum") String designNum, @PathVariable("author") String author) {
        return designRepository.findByDesignNumAndAuthor(designNum, author);
    }

    @RequestMapping(value = "/all" ,method = RequestMethod.GET, produces = "application/json")
    public List<Design> getAllDesigns(){
        return designRepository.findAll();
    }

    @RequestMapping(value = "/save" ,method = RequestMethod.POST, produces = "application/json")
    public List<Design> saveDesigns(@RequestBody List<Design> designs){
        for (Design design : designs) {
            designService.updateOrSaveDesign(design);
        }
       return  designs;
    }

}
